lib/matplotlib/tests/test_axes.py

```diff
@@ -1190,6 +1190,13 @@ def test_pcolorargs():
         ax.pcolormesh(x, y, Z[:-1, :-1], shading="gouraud")
     with pytest.raises(TypeError):
         ax.pcolormesh(X, Y, Z[:-1, :-1], shading="gouraud")
+    with pytest.raises(TypeError):
+        xx = x
```

Comment:
Why are you renaming `x`?  Do you mean to use `xx = x[:]`?

Reply:
Umm well my understanding of scope and context management is rudimentary.  If I change x in this with-block does it not change it for the next with-block?

Reply:
> If I change x in this with-block does it not change it for the next with-block?

It does. But it look suspicious, because `xx = x` makes `xx` alias of `x`, so it is a noop line.

Reply:
Oh `xx = x.copy()`?

Reply:
Actually, I don't need to rename this at all...